flow/layers/backdrop_filter_layer_unittests.cc

```diff
@@ -0,0 +1,188 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/flow/layers/backdrop_filter_layer.h"
+
+#include "flutter/flow/testing/layer_test.h"
+#include "flutter/flow/testing/mock_layer.h"
+#include "flutter/fml/macros.h"
+#include "flutter/testing/mock_canvas.h"
+#include "third_party/skia/include/core/SkImageFilter.h"
+#include "third_party/skia/include/effects/SkImageFilters.h"
+
+namespace flutter {
+namespace testing {
+
+static constexpr SkRect kEmptyRect = SkRect::MakeEmpty();
+
+using BackdropFilterLayerDeathTest = LayerTest;
+using BackdropFilterLayerTest = LayerTest;
+
+TEST_F(BackdropFilterLayerDeathTest, EmptyLayer) {
+  auto layer = std::make_shared<BackdropFilterLayer>(sk_sp<SkImageFilter>());
+
+  layer->Preroll(preroll_context(), SkMatrix());
+  EXPECT_EQ(layer->paint_bounds(), kEmptyRect);
+  EXPECT_FALSE(layer->needs_painting());
+  EXPECT_FALSE(layer->needs_system_composite());
+
+  EXPECT_DEATH_IF_SUPPORTED(layer->Paint(paint_context()),
+                            "needs_painting\\(\\)");
```

Comment:
Nit: Why the backslashes?

Reply:
EXPECT_DEATH treats the 2nd parameter as a POSIX regular expression, so you have to escape characters like . and ()

Reply:
Good to know. Thanks for clarifying.